wordpress-mobile/WordPress-Android
# Unified Dashboard: Refactoring Part 1

Part of #19270 

This PR starts refactoring for unified dashboard

- Removes the initial screen from the app setting view (does not remove getting/setting helper methods from AppPrefsWrapper or AppPrefs - this will be done separately)
- Renames MySiteTabFragment to MySiteMenuFragment
- Renames my_site_tab_fragment to my_site_menu_fragment
- Removes onCreateSiteResult

The Menu logic was extracted out into standalone classes with no reliance on source manager or mysiteviewmodel. The more tap will launch the new set of activities. The old classes were not removed and can be launched by removing the comment in MySiteFragment line 649.
- Adds `MenuActivity` and `MenuViewModel` into the `menu` package under MySite
- Enabled More tab to launch the MenuActivity instead of the MySiteMenuFragment

Notes
- The compose pieces need some tidying up/styling. 
- Quick start does not function the new MenuActivity. Focus points will work, but unless the activeTask is known, the builder won't set the focus point.
- Unit tests will come after if it is decided that we will continue with this approach
 
**To test:**
- Install the app and login
- Choose a site
- Navigate to My Site > More
- ✅ Verify the Menu view is displayed
- ✅ Verify all navigation actions work as normal

## Regression Notes
1. Potential unintended areas of impact
The more menu is not shown or working properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

UI Changes testing checklist:
